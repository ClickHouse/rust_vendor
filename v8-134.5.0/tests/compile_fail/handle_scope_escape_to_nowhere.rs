// Copyright 2019-2020 the Deno authors. All rights reserved. MIT license.

pub fn main() {
  let mut isolate = v8::Isolate::new(mock());
  let mut _scope = v8::EscapableHandleScope::new(&mut isolate);
}

fn mock<T>() -> T {
  unimplemented!()
}
