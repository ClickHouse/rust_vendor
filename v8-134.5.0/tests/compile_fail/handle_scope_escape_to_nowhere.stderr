error[E0277]: the trait bound `OwnedIsolate: v8::scope::param::NewEscapableHandleScope<'_, '_>` is not satisfied
 --> tests/compile_fail/handle_scope_escape_to_nowhere.rs:5:50
  |
5 |   let mut _scope = v8::EscapableHandleScope::new(&mut isolate);
  |                    ----------------------------- ^^^^^^^^^^^^ the trait `v8::scope::param::NewEscapableHandleScope<'_, '_>` is not implemented for `OwnedIsolate`
  |                    |
  |                    required by a bound introduced by this call
  |
  = help: the following other types implement trait `v8::scope::param::NewEscapableHandleScope<'s, 'e>`:
            `AllowJavascriptExecutionScope<'p, P>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'e>`
            `CallbackScope<'p, C>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'p>`
            `ContextScope<'p, P>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'e>`
            `DisallowJavascriptExecutionScope<'p, P>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'e>`
            `EscapableHandleScope<'p, 'e, C>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'p>`
            `HandleScope<'p, C>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'p>`
            `TryCatch<'p, P>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'e>`
note: required by a bound in `EscapableHandleScope::<'s, 'e>::new`
 --> src/scope.rs
  |
  |   pub fn new<P: param::NewEscapableHandleScope<'s, 'e>>(
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `EscapableHandleScope::<'s, 'e>::new`

error[E0277]: the trait bound `OwnedIsolate: v8::scope::param::NewEscapableHandleScope<'_, '_>` is not satisfied
 --> tests/compile_fail/handle_scope_escape_to_nowhere.rs:5:20
  |
5 |   let mut _scope = v8::EscapableHandleScope::new(&mut isolate);
  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `v8::scope::param::NewEscapableHandleScope<'_, '_>` is not implemented for `OwnedIsolate`
  |
  = help: the following other types implement trait `v8::scope::param::NewEscapableHandleScope<'s, 'e>`:
            `AllowJavascriptExecutionScope<'p, P>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'e>`
            `CallbackScope<'p, C>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'p>`
            `ContextScope<'p, P>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'e>`
            `DisallowJavascriptExecutionScope<'p, P>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'e>`
            `EscapableHandleScope<'p, 'e, C>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'p>`
            `HandleScope<'p, C>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'p>`
            `TryCatch<'p, P>` implements `v8::scope::param::NewEscapableHandleScope<'s, 'e>`
